File: tools/sdks/go/examples/quickstart_audit.go
//go:build ignore

// quickstart_audit — Stream audit events via GET /v1/admin/audit-events.
//
// Customer concept: every privileged op is Merkle-chained into the
// audit log (INV-AUDIT-APPEND-ONLY). Response includes chain_head_hash
// for client-side verification (CTRL-AUDIT-002).
//
// Run: CORELINK_PAT=$ADMIN_PAT go run examples/quickstart_audit.go
package main

import (
	"encoding/json"
	"fmt"
	"io"
	"net/http"
	"net/url"
	"os"
	"time"
)

func main() {
	api := os.Getenv("CORELINK_API_URL")
	if api == "" {
		api = "https://sandbox.corelink.dev"
	}
	pat := os.Getenv("CORELINK_PAT")
	if pat == "" {
		fmt.Fprintln(os.Stderr, "error: CORELINK_PAT env var is required")
		os.Exit(2)
	}
	u, err := url.Parse(api + "/v1/admin/audit-events")
	if err != nil {
		fmt.Fprintf(os.Stderr, "parse url: %v\n", err)
		os.Exit(1)
	}
	q := u.Query()
	q.Set("limit", "50")
	u.RawQuery = q.Encode()
	req, err := http.NewRequest("GET", u.String(), nil)
	if err != nil {
		fmt.Fprintf(os.Stderr, "build request: %v\n", err)
		os.Exit(1)
	}
	req.Header.Set("Authorization", "Bearer "+pat)
	req.Header.Set("Accept", "application/json")
	client := &http.Client{Timeout: 10 * time.Second}
	resp, err := client.Do(req)
	if err != nil {
		fmt.Fprintf(os.Stderr, "http: %v\n", err)
		os.Exit(1)
	}
	defer resp.Body.Close()
	raw, _ := io.ReadAll(resp.Body)
	var decoded map[string]any
	events, head := 0, "<none>"
	if err := json.Unmarshal(raw, &decoded); err == nil {
		if e, ok := decoded["events"].([]any); ok {
			events = len(e)
		}
		if h, ok := decoded["chain_head_hash"].(string); ok {
			head = h
		}
	}
	fmt.Printf("status=%d events=%d chain_head=%s\n", resp.StatusCode, events, head)
}

File: tools/sdks/python/examples/quickstart_audit.py
#!/usr/bin/env python3
"""quickstart_audit — Stream audit events via GET /v1/admin/audit-events.

Customer concept: every privileged operation is Merkle-chained into the
audit log (INV-AUDIT-APPEND-ONLY). The response includes chain_head_hash
for client-side verification (CTRL-AUDIT-002).

Run:
    CORELINK_PAT=$ADMIN_PAT python3 examples/quickstart_audit.py
"""

from __future__ import annotations

import json
import os
import sys
import urllib.error
import urllib.parse
import urllib.request


def main() -> int:
    api = os.environ.get("CORELINK_API_URL", "https://corelink-api.humangr.com")
    pat = os.environ.get("CORELINK_PAT")
    if not pat:
        print("error: CORELINK_PAT env var is required", file=sys.stderr)
        return 2

    qs = urllib.parse.urlencode({"limit": "50"})
    req = urllib.request.Request(
        url=f"{api}/v1/admin/audit-events?{qs}",
        method="GET",
        headers={"Authorization": f"Bearer {pat}", "Accept": "application/json"},
    )
    try:
        with urllib.request.urlopen(req, timeout=10) as resp:
            payload = json.loads(resp.read().decode("utf-8"))
            events = payload.get("events") if isinstance(payload, dict) else None
            count = len(events) if isinstance(events, list) else 0
            head = payload.get("chain_head_hash", "<none>") if isinstance(payload, dict) else "<none>"
            print(f"status={resp.status} events={count} chain_head={head}")
    except urllib.error.HTTPError as exc:
        print(f"http_error status={exc.code} body={exc.read().decode('utf-8', 'replace')}", file=sys.stderr)
        return 1
    except urllib.error.URLError as exc:
        print(f"network_error: {exc.reason}", file=sys.stderr)
        return 1
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
